revex/dfa.py

```diff
@@ -247,10 +248,28 @@ def _draw(self, full=False):  # pragma: no cover
         graph.add_node(invisible_start_node, attr_dict={'label': ''}, color='white')
         graph.add_edge(invisible_start_node, self.start, attr_dict={'label': ' start'})
 
-        nx.drawing.nx_agraph.write_dot(graph, '/tmp/foo_%s.dot' % id(graph))
-        os.system(
-            'dot -Tpng /tmp/foo_{0}.dot -o /tmp/foo_{0}.png'.format(id(graph)))
-        os.system('open /tmp/foo_{0}.png'.format(id(graph)))
+        if 'ipykernel' in sys.modules:
```

Comment:
For posterity, this is based on these three SO answers:
- http://stackoverflow.com/questions/11854847/display-an-image-from-a-file-in-an-ipython-notebook
- http://stackoverflow.com/questions/15411967/how-can-i-check-if-code-is-executed-in-the-ipython-notebook
- http://stackoverflow.com/questions/8220108/how-do-i-check-the-operating-system-in-python